vllm/forward_context.py

```diff
@@ -205,6 +205,10 @@ class ForwardContext:
 
     ubatch_slices: UBatchSlices | None = None
 
+    # set dynamically for each forward pass
+    # True during memory profiling, False otherwise
+    is_memory_profile: bool = False
+
```

Comment:
Should we avoid adding too many things to the forward_context? It is becoming increasingly complicated and I am increasingly worried about this class getting more and more bloated. cc @WoosukKwon @youkaichao